.github/workflows/go.yaml

```diff
@@ -73,5 +73,28 @@ jobs:
         working-directory: ${{ env.working-directory }}
         run: make docker-push
 
+  publish-helm:
```

Comment:
probably best to `if` on job level.

might worth to condition it on branch name to avoid republishing on periodics and in case we create `release-1.x` branches for the future that will also run CI.